### PR TITLE
feat(ts/analyzer): disallow `new` function target with noImplicitAny

### PR DESCRIPTION
**Pull request by littledivy**
```ts
// @noImplicitAny: true

function fnNumber(this: void): number { return 90; }
new fnNumber(); // Error
```

```
error[TS7009]: TargetLacksConstructSignature {
    span: Span {
        lo: BytePos(
            80,
        ),
        hi: BytePos(
            94,
        ),
        ctxt: #0,
    },
}
 --> a.ts:4:1
  |
4 | new fnNumber(); // Error
```

### CODE REVIEW

**Review by kdy1 — APPROVED** [review]
Can you rebase?

File: crates/stc_ts_errors/src/lib.rs [hunk 1]
```diff
@@ -342,11 +347,6 @@ pub enum Error {
         span: Span,
     },
 
-    /// TS18030
-    OptionalChainCannotContainPrivateIdentifier {
```
Comment (kdy1) [inline crates/stc_ts_errors/src/lib.rs:346]: Why did you remove this?

Comment (littledivy) [reply crates/stc_ts_errors/src/lib.rs:346]: Messed up the rebase 😅

**Review by kdy1 — APPROVED** [review]
